icu4c/source/common/uresdata.h

```diff
@@ -512,20 +514,27 @@ class ResourceDataValue : public ResourceValue {
 public:
     ResourceDataValue() :
         pResData(nullptr),
+        validLocaleDataEntry(nullptr),
         res(static_cast<Resource>(URES_NONE)),
         fTraceInfo() {}
     virtual ~ResourceDataValue();
 
     void setData(const ResourceData &data) {
         pResData = &data;
     }
+    
+    void setValidLocaleDataEntry(UResourceDataEntry *entry) {
+        validLocaleDataEntry = entry;
+    }
 
     void setResource(Resource r, ResourceTracer&& traceInfo) {
         res = r;
         fTraceInfo = traceInfo;
     }
 
     const ResourceData &getData() const { return *pResData; }
+    UResourceDataEntry *getValidLocaleDataEntry() const { return validLocaleDataEntry; }
```

Comment:
Question: Should this be const? When we return a non-const pointer, there is a logical conflict with making the getter const, but maybe it needs to be const so that call sites work?

(And the data entry pointer must be non-const so that init_resb_result() can entryIncrease() on it.)

Reply:
I started with it const and changed it because `getAliasTargetAsResourceBundle()` wanted it as a non-const and casting away the const didn't seem like a good idea.  Changing the parameter to `getAliasTargetAsResourceBundle()` to be const too just pushes the problem back to `init_resb_result()`, which is also expecting a non-const pointer.

>(And the data entry pointer must be non-const so that init_resb_result() can entryIncrease() on it.)

Not sure what you're saying here.  As I read this, that _is_ the reason why I didn't declare this member variable to be const.  Is there a better solution?

Reply:
Sorry I wasn't clear. The returned pointer must be non-const. I meant to ask *whether the **function** can be non-const*. As in
```
UResourceDataEntry *getValidLocaleDataEntry() { return validLocaleDataEntry; }
```
(`const` removed *in the middle* of the line)

If it's inconvenient for this function to be non-const (e.g., you would then need a const_cast) then leave it.
This is actually not really important, so you could leave this one anyway.